src/benchmarkstt/metrics/core.py

```diff
@@ -159,6 +160,157 @@ def compare(self, ref: Schema, hyp: Schema) -> OpcodeCounts:
 
 
+class BEER(Metric):
+    """
+    Bag of Entities Error Rate, BEER, is defined as the error rate per entity with a bag of words approach::
+
+                            abs(ne_hyp - ne_ref)
+        BEER (entity)   =   ----------------------
+                                ne_ref
+
+    - ne_hyp = number of detections of the entity in the hypothesis file
+    - ne_ref = number of detections of the entity in the reference file
+
+    The WA_BEER for a set of N entities is defined as the weighted average of the BEER for the set of
+    entities::
+
+        WA_BEER ([entity_1, ... entity_N) =  w_1*BEER (entity_1)*L_1/L + ... + w_N*BEER (entity_N))*L_N/L
+
+    which is equivalent to::
+
+                                            w_1*abs(ne_hyp_1 - ne_ref_1) + ... + w_N*abs(ne_hyp_N - ne_ref_N)
+        WA_BEER ([entity_1, ... entity_N) = ------------------------------------------------------------------
+                                                                            L
+
+    - L_1 =  number of occurrences of entity 1 in the reference document
+    - L = L_1 + ... + L_N
+
+    the weights being normalised by the tool:
+
+    - w_1 + ... + w_N = 1
+
+    The input file defines the list of entities and the weight per entity, w_n. It is processed as a json file with the
+    following structure::
+
+        { "entity_1":W_1, "entity_2" : W_2, "entity_3" :W_3 .. }
+
+    W_n being the non-normalized weight, the normalization of the weights is performed by the tool as::
+
+                    W_n
+        w_n =   ---------------
+                W_1 + ... +W_N
+
+    The minimum value for weight being 0.
+
+    """
+    def __init__(self, entities_file=''):
+        """
+        """
+        self._weight = []
+        self._entities = []
+
+        try:
+            with open(entities_file) as f:
+                data = json.load(f)
+            self._entities = list(data.keys())
+            weight = list(data.values())
+            self.set_weight(weight)
+        except IOError:
+            print('Input file Error')
+
+        return
+
+    def get_weight(self):
+        return self._weight
+
+    def set_weight(self, weight):
+        weight = [0 if w < 0 else w for w in weight]
+        sw = sum(weight)
+        if sw > 0:
+            self._weight = [w / sw for w in weight]
+
+    def get_entities(self):
+        return self._entities
+
+    def set_entities(self, entities):
+        self._entities = entities
+
+    # find the position of one entity
+    # an entity can contain more than one word
+    @staticmethod
+    def __find_pattern(search_list, complex_entity):
+        entity = ''.join(complex_entity).split(' ')
+        le = len(entity)
+        # complex_entity = [entity1 entity2 ...]
+        # the cursor sweep complex_entity to find consecutive entities entity1 ... entity2
+        cursor = 0
+        idx_found = []
+        for idx, elt in enumerate(search_list):
```

Comment:
This `idx` isn't being used. To avoid confusion with the definition of `idx` in line 252, line 248 can be simplified
```suggestion
        for elt in search_list:
```

Reply:
No, it is not equivalent. I renamed the var idx to idex_sl for sake of readability but it can not be removed see line 252

Reply:
You're right. Thanks for clarifying.